lib/pbench/test/unit/server/query_apis/commons.py

```diff
@@ -129,6 +131,24 @@ def get_expected_status(self, payload: JSON, header: HeaderTypes) -> int:
             expected_status = HTTPStatus.OK
         return expected_status
 
+    def make_request_call(
+        self, client, url, header: JSON, json: JSON = None, data=None
+    ):
+        if self.api_method == API_METHOD.GET:
```

Comment:
You could consider:

```
if self.api_method not in (API_METHOD.GET, API_METHOD.POST):
    raise Exception(f"Unexpected API method, {self.api_method}")
_func = client.get if self.api_method == API_METHOD.GET else client.post
return _func(url, headers=header, json=json, data=data)
```

Reply:
> Is there a reason not to consider this change?

Actually, I tried using the above code block as well, but while asserting and validating `data` and `json` only for the `GET` operation. This seemed to be a more simple way

Reply:
I like the `assert` rather than explicitly raising `Exception` since this is test code. Though it's definitely "debatable" since this is test infrastructure diagnosing a problem in the unit test code rather than a problem in the product code we're testing...